### ISSUE TITLE
ESPnet SSE Demonstration.ipynb: TypeError: Expected 'Iterator' ...

### ISSUE THREAD

**Issue by r9y9** [espnet/espnet#3428]
**Describe the bug**

ESPnet SSE Demonstration does not work on colab.

ESPnet SSE Demonstration.ipynb: https://colab.research.google.com/drive/1fjRJCh96SoYLZPRxsjF9VDv4Q2VoIckI?usp=sharing

I got errors while running the noteobok on colab:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-4-6a1109169182> in <module>()
      3 import sys
      4 import soundfile
----> 5 from espnet2.bin.enh_inference import SeparateSpeech
      6 
      7 

6 frames
/usr/local/lib/python3.7/dist-packages/torch/utils/data/_typing.py in _dp_init_subclass(sub_cls, *args, **kwargs)
    369                      return_hint.__origin__ == collections.abc.Iterator)):
    370                 raise TypeError("Expected 'Iterator' as the return annotation for `__iter__` of {}"
--> 371                                 ", but found {}".format(sub_cls.__name__, _type_repr(hints['return'])))
    372             data_type = return_hint.__args__[0]
    373             if not issubtype(data_type, sub_cls.type.param):

TypeError: Expected 'Iterator' as the return annotation for `__iter__` of IterableESPnetDataset, but found typing.Iterable[typing.Tuple[typing.Union[str, int], typing.Dict[str, numpy.ndarray]]]
```

Could anyone help to solve the issue?

**Basic environments:**

Google colab
 
**Environments from `torch.utils.collect_env`:*

```
Collecting environment information...
PyTorch version: 1.9.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.12.0
Libc version: glibc-2.26

Python version: 3.7 (64-bit runtime)
Python platform: Linux-5.4.104+-x86_64-with-Ubuntu-18.04-bionic
Is CUDA available: True
CUDA runtime version: 11.0.221
GPU models and configuration: GPU 0: Tesla T4
Nvidia driver version: 460.32.03
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.0.4
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] pytorch-wpe==0.0.1
[pip3] torch==1.9.0+cu102
[pip3] torch-complex==0.2.1
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.10.0
[pip3] torchvision==0.10.0+cu102
[conda] Could not collect
```

**Task information:**
 - Task: SE
 - Recipe: ESPnet SSE Demonstration.ipynb
 - ESPnet 2

**To Reproduce**

1. Open ESPnet SSE Demonstration.ipynb on colab
2. Run by "Enhance the single-channel real noisy speech in CHiME4"

**Comment by sw005320** [comment]
@LiChenda, could you deal with it?

**Comment by LiChenda** [comment]
This is caused by Pytorch 1.9, it is fixed in #3310 . I have updated the Colab notebook.

**Comment by r9y9** [comment]
Sorry for being late! I did check again and it worked! Thank you!
